test/local/routes/devices-and-sessions.js

```diff
@@ -498,6 +509,116 @@ describe('/account/devices/notify', function () {
   })
 })
 
+
+describe('/account/device/messages', function () {
+  const uid = uuid.v4('binary').toString('hex')
+  const deviceId = crypto.randomBytes(16).toString('hex')
+  const mockLog = mocks.mockLog()
+  const mockRequest = mocks.mockRequest({
+    log: mockLog,
+    devices: [
+      {
+        id: 'bogusid1',
+        type: 'mobile'
+      },
+      {
+        id: 'bogusid2',
+        type: 'desktop',
+      }
+    ],
+    credentials: {
+      uid: uid,
+      deviceId: deviceId
+    }
+  })
+  const mockPush = mocks.mockPush()
+  const mockCustoms = mocks.mockCustoms()
+
+  it('retrieves messages using the pushbox service', () => {
+    const pushboxToken = '0c2e73a6a72408f7c43ba3d48e207f24376935632125beb972c9a5da0899f855'
+    const mockOAuthMint = mocks.mockOAuthMint({
+      getOAuthToken: sinon.spy(() => {
+        return Promise.resolve({access_token: pushboxToken})
+      })
+    })
+    const mockPushbox = mocks.mockPushbox({
+      store: sinon.spy(() => {
+        return Promise.resolve({index: 'toktok'})
+      })
+    })
+    mockRequest.query = {
+      service: 'sendtab',
+      index: 2
+    }
+    const route = getRoute(makeRoutes({
+      customs: mockCustoms,
+      log: mockLog,
+      oAuthMint: mockOAuthMint,
+      pushbox: mockPushbox
+    }), '/account/device/messages', 'GET')
+
+    return runTest(route, mockRequest).then(() => {
+      assert.equal(mockPushbox.retrieve.callCount, 1, 'pushbox was called')
+      const args = mockPushbox.retrieve.args[0]
+      assert.equal(args.length, 5)
+      assert.equal(args[0], uid)
+      assert.equal(args[1], deviceId)
+      assert.equal(args[2], 'sendtab')
+      assert.deepEqual(args[3], {index: 2, limit: undefined})
+      assert.equal(args[4], pushboxToken)
+    })
+  })
+
+  it('store messages using the pushbox service', () => {
+    const pushboxToken = '558f9980ad5a9c279beb52123653967342f702e84d3ab34c7f80427a6a37e2c0'
+    const mockOAuthMint = mocks.mockOAuthMint({
+      getOAuthToken: sinon.spy(() => {
+        return Promise.resolve({access_token: pushboxToken})
+      })
+    })
+    const mockPushbox = mocks.mockPushbox({
+      store: sinon.spy(() => {
```

Comment:
would it be useful to have a test for when pushbox fails to store or retrieve, to make sure error response is correct?

Reply:
So far we're only echo-ing back the response we receive from the server, I think this is fine.

Reply:
FWIW I think it'd be worth at least one test just to verify that that works as expected in the error case :-)

Reply:
It looks like there are still some untested error cases here:

* Unknown device id in "to"
* Unknown topic in "topic"
* Operational error from pushbox service